Modules/_asynciomodule.c

```diff
@@ -962,14 +962,18 @@ FutureObj_dealloc(PyObject *self)
 {
     FutureObj *fut = (FutureObj *)self;
 
+    PyObject_GC_UnTrack(self);
+
     if (Future_CheckExact(fut)) {
         /* When fut is subclass of Future, finalizer is called from
          * subtype_dealloc.
          */
+         PyObject_GC_Track(self);
```

Comment:
> It seems like every dealloc should have a PyObject_GC_UnTrack at the top?

When the type has `Py_TPFLAGS_HAVE_GC`.
`lru_cache_dealloc` in `_functoolsmodule` seems need it.